bin/beepmein

```diff
@@ -0,0 +1,281 @@
+# On Debian systems make sure 'at' is installed with --no-install-recommends.
+
+# Copyright (C) 2019/2020 brontosaurusrex
+# License: GPLv3 or later.
+
+# prepare config dir
+config="$HOME/.cache/beepmein"
+mkdir -p "$config"
+
+# var
+delay="0"
+
+# softlinks to audio samples
+alarmSample="$config/alarm"     # longer sample
+reminderSample="$config/remind" # shorter sample
+
+# Will kill all children with 'killall beepmein'
+trap 'kill $(jobs -p) >/dev/null 2>&1' EXIT
+
+# required
+command -v at >/dev/null 2>&1 || { >&2 echo "I need at: sudo apt install --no-install-recommends at" ; exit 1; }
+command -v speaker-test >/dev/null 2>&1 || { >&2 echo "I need speaker-test from alsa-utils (for reminder sound)." ; exit 1; }
+command -v notify-send >/dev/null 2>&1 || { >&2 echo "I need notify-send." ; exit 1; }
+#( command -v mpv >/dev/null 2>&1 || command -v cvlc >/dev/null 2>&1 ) || { >&2 echo "I need mpv or vlc (for alarm sounds)." ; exit 1; }
+command -v yad >/dev/null 2>&1 || { >&2 echo "I need yad." ; exit 1; }
+
+# help
+help () { 
+cat << EOF
+beepmein help
+
+    Set reminder or alarm:
+
+        beepmein now + 3 minutes # reminder ('at' time format)
+        beepmein 22:10           # reminder ('at' time format)
+        beepmein 3               # reminder (in 3 minutes, beepmein shortcut),
+        beepmein 8:00 --alarm    # alarm (must be last parameter)
+        beepmein --nr 5          # with no reason (must be first parameter)
+           
+    Set sounds: (playback depends on mpv or vlc)
+
+        beepmein --set-alarm <path/to/alarm.sound>
+        beepmein --set-remind <path/to/reminder.sound>
+
+    Test playback, blinking and notifications:
+
+        beepmein --reaction reminder_text
+        beepmein --reaction --alarm alarm_text
+        beepmein --test # test everything
+        
+    List jobs
+    
+        beepmein --list # or -l
+
+    Kill all running playback/beeping:
+
+        killall beepmein 
+
+    Notes:
+
+        at -l       # will list pending jobs (or atq)
+        at -r <num> # will remove job <num>  (or atrm)
+
+EOF
+}
+
+# superecho
+superecho () {
+    if [ -t 0 ] ; then
+        echo "$1"
+    else
+        notify-send -t 3500 "$1"
+    fi
+}
+
+# reminder sound, needs speaker-test (alsa-utils)
+beep () {
+    [ -z "$1" ] && loops="1" || loops="$1"
+    for ((i=1; i<=loops; i++))
+    do
+        timeout -s HUP 0.2 speaker-test -t sine -f 1000  > /dev/null 2>&1
+        sleep 0.2
+    done
+}
+
+# $1 = how many times to blink, $2 = pause between each blink
+blink () {
+    # find primary connected or else 1st connected monitor
+    monitor="$(xrandr -q | grep -w "connected" | grep "primary" | cut -f1 -d ' ')"
+    if [ -z "$monitor" ]; then # no primary AND connected found, lets try 1st connected
+        monitor="$(xrandr -q | grep -w "connected" | cut -f1 -d ' ')"
+    fi 
+    [ -z "$monitor" ] && { >&2 echo "No monitors found error" ; exit 1; }
```

Comment:
Might want to introduce a function `die`.

```bash
die() {
local msg=$1
printf "%s\n" "$msg"
exit 1
}
```
or so.

Reply:
For the purpose of? (i don't understand)

Reply:
You fail at multiple places by echoeing a message and then exiting. I just thought perhaps you could short all these cases to

```
somecommand || die "message"
```

instead of typing out the entire Bash block {} each time.